R/remote_api.R

```diff
@@ -6,29 +6,41 @@ pull_archive_api <- function(name, id, config, remote) {
   loadNamespace("montagu")
   assert_is(config, "orderly_config")
   orderly_remote_resolve_secrets(config, remote)
+
+  v <- withCallingHandlers(
+    montagu::montagu_reports_report_versions(name, remote),
+    error = function(e) {
+      valid <- montagu::montagu_reports_list(remote)
+      if (!(name %in% valid$name)) {
+        stop(sprintf("No versions of report '%s' found at '%s'",
+                     name, remote$name),
+             call. = FALSE)
+      }
+    })
+
   if (id == "latest") {
     ## Resolve id
     v <- montagu::montagu_reports_report_versions(name, remote)
```

Comment:
Shouldn't this be the same as `v` on line 10? Is there something subtle going on with `withCallingHandlers`?

Reply:
you're completely right - that was redundant